### PR TITLE
HBASE-23699 ByteBuff may not recycle correctly when NettyRpcServer enabled and client disconnect

### PR DESCRIPTION
**Pull request by chenxu14**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 35s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 11s |  master passed  |
| +1 :green_heart: |  compile  |   1m 18s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 44s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 33s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 58s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 31s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   5m 18s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 52s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 20s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 20s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 43s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 36s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  15m 32s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 56s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   5m 23s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 20s |  hbase-common in the patch passed.  |
| -1 :x: |  unit  | 242m  6s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m  8s |  The patch does not generate ASF License warnings.  |
|  |   | 308m 21s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.util.TestMiniClusterLoadParallel |
|   | hadoop.hbase.replication.TestReplicationSmallTests |
|   | hadoop.hbase.client.TestAsyncTableGetMultiThreaded |
|   | hadoop.hbase.master.assignment.TestRegionMoveAndAbandon |
|   | hadoop.hbase.client.TestMultiParallel |
|   | hadoop.hbase.io.encoding.TestLoadAndSwitchEncodeOnDisk |
|   | hadoop.hbase.client.TestMultiRespectsLimits |
|   | hadoop.hbase.coprocessor.TestMetaTableMetrics |
|   | hadoop.hbase.client.TestMobRestoreSnapshotFromClientAfterSplittingRegions |
|   | hadoop.hbase.client.TestClientOperationInterrupt |
|   | hadoop.hbase.master.procedure.TestSCPWithMetaWithoutZKCoordinated |
|   | hadoop.hbase.master.assignment.TestMergeTableRegionsProcedure |
|   | hadoop.hbase.client.TestCloneSnapshotFromClientCloneLinksAfterDelete |
|   | hadoop.hbase.util.TestMiniClusterLoadEncoded |
|   | hadoop.hbase.master.procedure.TestRestoreSnapshotProcedure |
|   | hadoop.hbase.master.procedure.TestMasterFailoverWithProcedures |
|   | hadoop.hbase.client.TestAsyncNonMetaRegionLocator |
|   | hadoop.hbase.security.visibility.TestVisibilityLabelsWithACL |
|   | hadoop.hbase.util.TestMiniClusterLoadSequential |
|   | hadoop.hbase.quotas.TestQuotaObserverChoreWithMiniCluster |
|   | hadoop.hbase.quotas.TestSpaceQuotasWithRegionReplicas |
|   | hadoop.hbase.regionserver.TestJoinedScanners |
|   | hadoop.hbase.client.TestAsyncTable |
|   | hadoop.hbase.client.TestMobRestoreSnapshotFromClientAfterTruncate |
|   | hadoop.hbase.regionserver.TestHRegionServerBulkLoad |
|   | hadoop.hbase.snapshot.TestRestoreFlushSnapshotFromClient |
|   | hadoop.hbase.quotas.TestQuotaStatusRPCs |
|   | hadoop.hbase.TestMultiVersions |
|   | hadoop.hbase.client.TestRestoreSnapshotFromClientSimple |
|   | hadoop.hbase.client.TestFromClientSide3 |
|   | hadoop.hbase.master.procedure.TestTruncateTableProcedure |
|   | hadoop.hbase.quotas.TestSpaceQuotaBasicFunctioning |
|   | hadoop.hbase.quotas.TestSpaceQuotaSwitchPolicies |
|   | hadoop.hbase.quotas.TestSpaceQuotaRemoval |
|   | hadoop.hbase.master.TestMasterNoCluster |
|   | hadoop.hbase.util.TestFromClientSide3WoUnsafe |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1055/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1055 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 02e8f9222dd3 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1055/out/precommit/personality/provided.sh |
| git revision | master / 19d3bed1d4 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1055/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1055/1/testReport/ |
| Max. process+thread count | 5740 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1055/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcServerResponseEncoder.java [hunk 1]
```diff
@@ -44,7 +44,6 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
       RpcResponse resp = (RpcResponse) msg;
       BufferChain buf = resp.getResponse();
       ctx.write(Unpooled.wrappedBuffer(buf.getBuffers()), promise).addListener(f -> {
-        resp.done();
```
Comment (saintstack) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcServerResponseEncoder.java]: When is 'done' called or is it that we  now defer to cleanup?

Comment (saintstack) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcServerResponseEncoder.java]: Any chance of an explanation on how this addresses the issue? Thanks.

**Comment by chenxu14** [comment]
the issuse is our own branch only, sorry for the Noise, close it.

**Pull request closed without merging**